### DISCUSSION TITLE
Please confirm assumption on "centralPackageVersions"

### DISCUSSION THREAD

**Discussion by cz-dev-ge** [NuGet/Home#13240] [Q&A]
Hi there,
according to [the NuGet wiki](https://github.com/NuGet/Home/wiki/Centrally-managing-NuGet-package-versions#will-a-central-defined-package-version-influence-transitive-dependency-resolution) adding a package version to `Directory.Packages.props` enforces the same package version in the whole solution - even when not explicitly referenced in a project (lets assume it is not referenced anywhere in a project's explicit dependency tree - only via transitive dependencies).

ProjectA > PackageA (explicit) > PackageB v1.0.0 (transitive)
ProjectB > PackageC (explicit) > PackageB v1.4.0 (transitive)
Directory.Packages.props: PackageB v2.0.0

So the one actually used will be PackageB v2.0.0 - although not explicitly referenced in my solution. Right?

Is this what this means in the [Microsofts docs](https://learn.microsoft.com/en-us/nuget/consume-packages/central-package-management#transitive-pinning)?

> You can automatically override a transitive package version even without an explicit top-level &lt;PackageReference /&gt; by opting into a feature known as transitive pinning. This promotes a transitive dependency to a top-level dependency implicitly on your behalf when necessary.

So when I look into my `project.assets.json` in a built project and the tree resolves to PackageB v1.4.0 but the section `centralPackageVersions` looks like the following, version 2.0.0 will be used in the end, right?

```json
"centralPackageVersions": {
  "PackageB": "[2.0.0, )"
}
```

Best regards,
Georg

**Comment by zivkan** [comment]
The "wiki" is an early design spec, and we don't maintain design specs after new features modify the original spec's design, or even if the implementation discovered difficulties and therefore the design changed slightly. Only use learn.microsoft.com/nuget as a reference.

As the docs say, you have to opt into transitive pinning. `PackageVersion` will only apply to direct package references otherwise. Note, if you pack your own projects into packages, transitive pinning will cause the pinned packages to become direct dependencies of your package, which is why transitive pinning is not on by default.
